pyne/partisn.py

```diff
@@ -0,0 +1,816 @@
+    f.write("ievt=0      \ source calculation\n")
+    f.write("\ isct=     \ Legendre order of scattering (default=0)\n")
+    f.write("\ ith=      \ 0/1/2= direct/adjoint/POI calculation (default=0)\n")
+    f.write("\ ibl=      \ left BC (default=0, vacuum)\n")
+    f.write("\ ibr=      \ right BC (default=0, vacuum)\n")
+    f.write("\ ibt=      \ top BC (default=0, vacuum)\n")
+    f.write("\ ibb=      \ bottom BC (default=0, vacuum)\n")
+    f.write("\ ibfrnt=   \ front BC (default=0, vacuum)\n")
+    f.write("\ ibback=   \ back BC (default=0, vacuum)\n")
+    f.write("\ \n")
+    
+    f.write("\ Source is in format of option 3 according to PARTISN input manual.\n")
+    f.write("\ Default is an evenly distributed volume source.\n")
+    f.write("source= ")
+    count = 0
+    tot = 0
+    for row in block05['source']:
+        formatted_string = format_repeated_vector(row)
+        f.write(formatted_string)
+        tot += 1
+        count += 1
+        if count == 4:
+            if tot != len(block05['source']):
+                f.write(";\n        ")
+            else:
+                f.write(";")
+            count = 0
+        else:
+            f.write("; ")
+    f.write("\n")
+        
+    if 'sourcx' in block05.keys():
+        f.write("sourcx= ")
+        count = 0
+        tot = 0
+        for row in block05['sourcx']:
+            formatted_string = format_repeated_vector(row)
+            f.write(formatted_string)
+            tot += 1
+            count += 1
+            if count == 4:
+                if tot != len(block05['sourcx']):
+                    f.write(";\n        ")
+                else:
+                    f.write(";")
+                count = 0
+            else:
+                f.write("; ")
+        f.write("\n")
+                
+    if 'sourcy' in block05.keys():
+        f.write("sourcy= ")
+        count = 0
+        tot = 0
+        for row in block05['sourcy']:
+            formatted_string = format_repeated_vector(row)
+            f.write(formatted_string)
+            tot += 1
+            count += 1
+            if count == 4:
+                if tot != len(block05['sourcy']):
+                    f.write(";\n        ")
+                else:
+                    f.write(";")
+                count = 0
+            else:
+                f.write("; ")
+        f.write("\n")
+            
+    if 'sourcz' in block05.keys():
+        f.write("sourcz= ")
+        count = 0
+        tot = 0
+        for row in block05['sourcz']:
+            formatted_string = format_repeated_vector(row)
+            f.write(formatted_string)
+            tot += 1
+            count += 1
+            if count == 4:
+                if tot != len(block05['sourcz']):
+                    f.write(";\n        ")
+                else:
+                    f.write(";")
+                count = 0
+            else:
+                f.write("; ")
+        f.write("\n")
```

Comment:
So this is a somewhat unsafe way to create files.  It is much better to create the full string representation in memory ahead of time and then call `write()` once and only once at the end of the function.  In this way you don't risk accidentally writing part of file. Also, it is faster because you are not constantly hitting the hard disk.